pygments/lexers/html.py

```diff
@@ -206,22 +206,26 @@ class XmlLexer(RegexLexer):
 
     tokens = {
         'root': [
-            ('[^<&]+', Text),
+            (r'[^<&\n]+', Text),
+            (r'\n\s*', Whitespace),
             (r'&\S*?;', Name.Entity),
-            (r'\<\!\[CDATA\[.*?\]\]\>', Comment.Preproc),            
+            (r'\<\!\[CDATA\[.*?\]\]\>', Comment.Preproc),
             (r'<!--(.|\n)*?-->', Comment.Multiline),
             (r'<\?.*?\?>', Comment.Preproc),
             ('<![^>]*>', Comment.Preproc),
-            (r'<\s*[\w:.-]+', Name.Tag, 'tag'),
-            (r'<\s*/\s*[\w:.-]+\s*>', Name.Tag),
+            (r'(<)(\s*)([\w:.-]+)',
+                bygroups(Punctuation, Whitespace, Name.Tag), 'tag'),
+            (r'(<)(\s*)(/)(\s*)([\w:.-]+)(\s*)(>)',
+                bygroups(Punctuation, Whitespace, Punctuation, Whitespace,
+                    Name.Tag, Whitespace, Punctuation)),
```

Comment:
I was thinking, that tags very often contain arguments. So tags are broken into separate tokens anyway.
But I totally agree, this is to be discussed. I am not saying that either way would be better since i did not recognize a "design" across languages yet.

Reply:
FWIW, Firefox for instance highlights tags the same way as the HTML output today (not like XML), but Chromium does it the other way round (like our XML.) I would prefer it to be consistent between XML and HTML. I personally prefer the first style, especially when it comes to readability because it doesn't turn a long sequence of `</span></p></br>` into a solid color soup.

Reply:
OK, after looking at a real-world example with a few different styles, I have to take back what I said. Including the brackets makes the `<span` part of `<span attr="..." otherattr="..." lots of stuff ... >` look asymmetric, and as Matthaüs says, it collapses everything together for adjacent tags. I think the option proposed in this patch looks really good with a bold style for Name.Tag, and slightly less good (though still good) when the tags are just in a different color, but it's worth remembering that the default style uses bold for tags.